src/Control/FunFlow/External/Docker.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Control.FunFlow.External.Docker where
+
+import           Control.FunFlow.ContentHashable
+import           Control.FunFlow.External
+import           Data.Map.Strict                 (Map)
+import qualified Data.Map.Strict                 as Map
+import           Data.Semigroup                  ((<>))
+import qualified Data.Text                       as T
+import           GHC.Generics                    (Generic)
+import           System.FilePath
+
+data Bind
+  -- | Single input, will get mounted to @/input@ on the image.
+  = SingleInput ContentHash
+  -- | Multiple inputs, each gets mouted into a subdirectory under
+  -- @/input@ as described by the given mapping.
+  | MultiInput (Map FilePath ContentHash)
+  deriving Generic
+
+instance ContentHashable Bind
+
+data Config = Config
+  { image :: T.Text
+  , input :: Bind
```

Comment:
I wonder if we should include an optional image hash in the config, so that people can pin the container if desired.